Спринт №9

LasvegasandFlorida opened the pull request.

Commit:
Первый коммит dev

Commit:
Доработка dev

Commit:
Работа над спринтом №9

Commit:
Работа над спринтом №9 (ДЗ 2)

Commit:
Работа над спринтом №9 (ДЗ 4)

nadzyathedude commented on app/src/main/java/com/example/playlistmakerprod/MainActivity.kt:
```diff
@@ -1,11 +1,32 @@
+                startActivity(searchIntent)
+            }
+        }
+        searchBtn.setOnClickListener(searchBtnClickListener)
```
предлагаю не усложнять а просто сделать работу с листенером по аналогии с другими кнопками, как ты сделал ниже

LasvegasandFlorida replied:
Сделаю. Просто в ДЗ было указано сделать двумя способами.

nadzyathedude commented on app/src/main/java/com/example/playlistmakerprod/SettingsActivity.kt:
```diff
@@ -13,5 +17,42 @@ class SettingsActivity : AppCompatActivity() {
+            shareIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.my_email)))
+            shareIntent.putExtra(Intent.EXTRA_TEXT, getString(R.string.mail_body))
+            startActivity(shareIntent)
+        }
```
вышла немного громосткая конструкция. предлагаю подсократить, например так

```
 val supportBtn = findViewById<Button>(R.id.support_button)
        supportBtn.setOnClickListener {
          startActivity(Intent(Intent.ACTION_SENDTO).apply {
                data = Uri.parse("mailto:")
                putExtra(Intent.EXTRA_SUBJECT, getString(R.string.mail_subject))
                putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.my_email)))
                putExtra(Intent.EXTRA_TEXT, getString(R.string.mail_body))
            }
          )
        }
```

nadzyathedude commented on app/src/main/java/com/example/playlistmakerprod/SearchActivity.kt:
```diff
@@ -1,11 +1,66 @@
+        clearButton.setOnClickListener {
+            inputEditText.setText("")
+            inputMethodManager?.hideSoftInputFromWindow(inputEditText.windowToken, 0)
+            inputEditText.clearFocus()
```
предлагаю вынести в отдельную функцию тк это другое состояние экрана. Назвать можно к примеру `onClearClick()`

Commit:
- Добавлено наследование кнопок на "Главном экране"
- Упращен код перехода на экран "Поиска"
- Упращен код кнопки отправки письма на экране "Настройки"
- Код очистки строки поиска на экране "Поиска" вынесен в отдельную функцию
